crates/miden-lib/asm/agglayer/account_components/bridge_out.masm

```diff
@@ -0,0 +1,215 @@
+use.miden::active_note
+use.miden::native_account
+use.miden::note
+use.miden::output_note
+use std::crypto::hashes::keccak256
+use std::word
+
+# CONSTANTS
+# =================================================================================================
+const.MMR_PTR=42
+const.STORAGE_MAP_INDEX=0
+
+const.BURN_NOTE_ROOT=[6407337173854817345,5626358912819151014,703918618794810515,17401169215223723177]
+const.EXECUTION_HINT_ALWAYS=1
+const.PUBLIC_NOTE=1
+const.AUX=0
+const.NUM_BURN_NOTE_INPUTS=0
+const.BURN_ASSET_MEM_PTR=24
+
+#! Computes the SERIAL_NUM of the outputted BURN note.
+#!
+#! The serial number is computed as hash(B2AGG_SERIAL_NUM, ASSET).
+#!
+#! Inputs:  [ASSET]
+#! Outputs: [SERIAL_NUM]
+#!
+#! Where:
+#! - ASSET is the asset for which to compute the burn note serial number.
+#! - SERIAL_NUM is the computed serial number for the BURN note.
+#!
+#! Invocation: exec
+proc compute_burn_note_serial_num
+    exec.active_note::get_serial_number
+    # => [B2AGG_SERIAL_NUM, ASSET]
+
+    hmerge
+    # => [SERIAL_NUM]
+end
+
+#! Creates a BURN note for the specified asset.
+#!
+#! This procedure creates an output note that represents a burn operation for the given asset.
+#! The note is configured with the appropriate recipient, tag, and execution hint.
+#!
+#! Inputs:  [ASSET]
+#! Outputs: []
+#!
+#! Where:
+#! - ASSET is the asset to be burned.
+#!
+#! Invocation: exec
+proc create_burn_note.8
```

Comment:
Not for this PR (and somewhat related to one of my previous comment): ideally, note constructors would live in the same package as the note's script. This way, getting note's script root should be relatively trivial (using `procref` instruction).

To do this, we'll need to refactor things a bit. Specifically:

- Note script should be stored in a library (in the future just package) where one of the procedures is the note's script, and other procedures are note constructors.
- We'll need a way to identify which procedure is a script and which ones are constructors. This could be done either via procedure attributes or package metadata.

Let's create an issue for this.